src/lib/storage/disk_size.rb

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "pp"
+
+# This file can be invoked separately for minimal testing.
+
+module Yast
+  module Storage
+    #
+    # Class to handle disk sizes in the MB/GB/TB range with readable output.
+    #
+    class DiskSize
+      include Comparable
+
+      UNITS = ["kiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"]
+      UNLIMITED = "unlimited"
+
+      attr_accessor :size_k
+
+      def initialize(size_k = 0)
+        @size_k = size_k.round
+      end
+
+      #
+      # Factory methods
+      #
+      class << self
+        def kiB(size)
+          DiskSize.new(size)
+        end
+
+        def MiB(size)
+          DiskSize.new(size * 1024)
+        end
+
+        def GiB(size)
+          DiskSize.new(size * (1024**2))
+        end
+
+        def TiB(size)
+          DiskSize.new(size * (1024**3))
+        end
+
+        def PiB(size)
+          DiskSize.new(size * (1024**4))
+        end
+
+        def EiB(size)
+          DiskSize.new(size * (1024**5))
+        end
+
+        def ZiB(size)
+          DiskSize.new(size * (1024**6))
+        end
+
+        def YiB(size)
+          DiskSize.new(size * (1024**7))
+        end
+
+        def unlimited
+          DiskSize.new(-1)
+        end
+
+        def zero
+          DiskSize.new(0)
+        end
```

Comment:
I just tried converting them to constants; then I can non longer write

    my_size1 = DiskSize.GiB(200)
    my_size2 = DiskSize.unlimited
    my_size3 = DiskSize.TiB(2)
    my_size4 = DiskSize.zero

I have to write

    my_size1 = DiskSize.GiB(200)
    my_size2 = DiskSize::UNLIMITED
    my_size3 = DiskSize.TiB(2)
    my_size4 = DiskSize::ZERO

The inconsistency between . and :: is yet another unnecessary cause for errors. In Ruby that's a lot worse because they only get detected during runtime.